framework/axSGreaseSkeleton.js

```diff
@@ -11,6 +11,59 @@
 
 /*** Functions for common tweaks. ***/
 
+/**
+ * Adds text to the given live region, and clears it a second later so it's no
+ * longer in the virtual buffer.
+ * @param {string} regionid an id of a region.
+ */
+function announce(text, regionId) {
+	getLiveRegion(regionId)
+		.then((region) => {
+			region.innerText = text;
+			setTimeout(() => {
+				region.innerText = '';
+			}, 1000);
+		});
+}
+
+/**
+ * create or fetch a live region that can be used with updateLiveRegion. Returns a promise with the region.
+ * @param {string} id the name of the new live region. This is an html id.
+ * @return {!Promise<HTMLElement>} a div that contains the live region. This can typically be ignored, this exxists to aid in chaining creation of non-existant regions.
+ */
+
+function getLiveRegion(id) {
+	const updatePromise = new Promise((resolve, reject) => {
+		if (!id) {
+			reject('Need a valid id!');
+			return;
+		}
+		const existingRegion = document.getElementById(id);
+		if (existingRegion) {
+			resolve(existingRegion);
+			return;
+		}
+		const region = document.createElement('div');
+		region.id = id;
+		region.setAttribute('aria-live', 'polite');
+		region.setAttribute('aria-atomic', 'true');
+		region.style.position = 'absolute';
+		region.style.width = '50px';
+		region.style.height = '50px';
+		region.style.opasity = 0;
+		document.body.appendChild(region);
+		// we need to delay a little to get the new region to actually read contents.
+		// A11y APIs probably don't treat the relevant changes as "additions" until
+		//an annimation frame has passed. It may, in reality be more like 2-4
+		// annimation frames, so delay 134 ms to be safe.
+		setTimeout(() => {
+			resolve(region);
+		}, 134);
+	});
+	return updatePromise;
+}
+
+/**
```

Comment:
Um... I think this comments out a huge block of code below?

Reply:
It's just the dangling start of a sad comment that was 99% deleted. I've gone back in and eliminated the rest of it. :-)